CommonTools/Utils/interface/LazyConstructed.h

```diff
@@ -0,0 +1,57 @@
+#ifndef CommonTools_Utils_LazyConstructed_h
+#define CommonTools_Utils_LazyConstructed_h
+// -*- C++ -*-
+//
+// Package:     CommonTools/Utils
+// Class  :     LazyConstructed
+//
+/**\class LazyConstructed LazyConstructed.h "CommonTools/Utils/interface/LazyConstructed.h"
+ Description: Wrapper around a class for lazy construction.
+
+ Usage:
+    // example: lazy SoA table
+    auto object = makeLazy<edm::soa::EtaPhiTable>(trackCollection);
+
+Notes:
+  * See similar class CommonTools/Utils/interface/LazyConstructed.h for implementation details.
```

Comment:
This is not a "similar class", it is the very same class: maybe you intended to point out to some different one?

Reply:
Yes, thanks, I wanted to point to `LazyResult.h`, which provides equivalent functionality for function evaluation.

Reply:
Since I already had to do the rebase and this change was trivial, I amended it to the "add LazyConstructed" commit.